packages/module-utils/test/generate-dts.spec.ts

```diff
@@ -247,6 +276,75 @@ describe('Generate DTS', function () {
                 "Type 'true' is not assignable to type 'string | undefined'"
             );
         });
+
+        describe('state overrides', () => {
+            it('should support states overridden through a local extend', () => {
+                tk.populate({
+                    'test.st.css':
+                        '.base { -st-states: state1; } .test { -st-states: state1(string); -st-extends: base; }',
+                    'test.ts': `
+                        import { eq } from "./test-kit";
+                        import { st, classes } from "./test.st.css";
+                        
+                        eq<string>(st(classes.test, { state1: 'hello' }));
+                    `,
+                });
+
+                expect(tk.typecheck('test.ts')).to.equal('');
+            });
+
+            it('should warn about incorrect state value that is overridden through a local extend', () => {
```

Comment:
Why do we warn about this ? seems legit

Reply:
I know this will make the types more complicated but we might want to solve it. (related to states per class)

Reply:
rephrased the `it`, hopes it's more descriptive.